docs/blog/posts/2023/ansible-srlinux-1.md

```diff
@@ -0,0 +1,947 @@
+  ```
+The _l2vpn_ role will transform this _fabric-wide_ intent into a node-specific intent per resource (network-instance, subinterface, tunnel-interface) and will merge this with the global node intent.
+
+The _l3vpn_ role follows a similar approach but depends on the _l2vpn_ role to define the intent for the mac-vrf instances. If not, the playbook will fail. The _l3vpn_ role knows if an ip-vrf instance applies to the node based of the mac-vrf definitions associated with the ip-vrf. The mac-vrf definition in the L2VPN intent includes the node association.
+
+
+An example of a L3VPN intent is shown below:
+
+```yaml title="roles/services/l3vpn/vars/test/l3vpn.yml"
+l3vpn:                      # root of l3vpn intent, mapping of ip-vrf instances, with key=ip-vrf name
+  ipvrf-2001:               # name of the ip-vrf instance
+    id: 2001                # id of the ip-vrf instance: used for route-targets
+    type: ip-vrf
+    description: IPVRF1
+    snet_list:              # a list of (macvrf, gw) pairs. The macvrf must be present in the l2vpn intent  
+      - macvrf: macvrf-300  # the macvrf instance to associate with the ip-vrf instance
+        gw: 10.1.1.254/24   # the gateway address for the subnet
+      - macvrf: macvrf-301
+        gw: 10.1.2.254/24
+    export_rt: 100:2001     # export route-target for EVPN address-family (route-type: 5)
+    import_rt: 100:2001     # import route-target for EVPN address-family (route-type: 5)
+```
+
+#### COMMON and UTILS roles
+
+Once the nodal intent has been constructed by the INFRA and SERVICES roles, the playbook calls the `common/configure` role as the last task. This role will take the nodal intent and construct the final configuration for the node. It calls roles in the `utils` folder to construct the configuration for the various resources (interfaces, network-instances, policies, etc) and thus generates the variables `update` and `replace` that are passed as arguments to the `nokia.srlinux.config` module.
+
+It also generates a `delete` variable containing a list of configuration paths to delete when the play variable `purge=true` and when no tags are specified with the `ansible-playbook` command that would result in a partial nodal intent. It uses the node for configuration state (running configuration) that was retrieved by the `common/init` role and compares this against the nodal intent to generate the `delete` variable.
+
+Following diagram gives an overview how the low-level device intent is constructed from the various roles:
+
+<figure markdown>
+  <div class="mxgraph" style="max-width:100%;border:1px solid transparent;margin:0 auto; display:block;" data-mxgraph="{&quot;page&quot;:0,&quot;zoom&quot;:2,&quot;highlight&quot;:&quot;#0000ff&quot;,&quot;nav&quot;:true,&quot;check-visible-state&quot;:true,&quot;resize&quot;:true,&quot;url&quot;:&quot;https://raw.githubusercontent.com/wdesmedt/ansible-srl-demo/main/img/ansible-srl-intent.drawio.svg&quot;}"></div>
```

Comment:
@wdesmedt here is how I embed vector graphics from diagrams.net.

In this ugly div you are only concerned about three parameters:

1. **url**: at the end of this div you see the URL that is a raw github link to the original file you used in drawio editor. Make sure to point it to the right location.
1. **page**: inside the `data-mxgraph` property you see the first element that goes like `&quot;page&quot;:0`. This is html-escaped version of `page:0`. `Page:0` means that from the referenced drawio file you render first page (aka tab), they are 0-indexed.
2. **zoom**: the last property you may want to touch here is the `zoom`, which is set to `2` in this example. Zoom is in charge of making small pictures larger if desired. You can play with this param to see how it changes the diagram.

Reply:
PS. I changed the URL escaping to a single-quotes escaping, which is much nicer now.